### PR TITLE
Customizable user namespace templating

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
This change allows a user to explicitly specify the templating used for workflow execution namespaces (which is then further customizable by the cluster resource controller). Because it was previously undocumented I don't think it's worth carrying over backwards compatibility with the existing enum. I'll work with L5 to make sure they're aware of the changes when they're released since it will affect them.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1265

## Follow-up issue
_NA_

### CODE REVIEW

**Review by wild-endeavor — APPROVED** [review]

File: flyteadmin_config.yaml [hunk 1]
```diff
@@ -200,3 +200,5 @@ qualityOfService:
     development: LOW
     staging: MEDIUM
     # by default production has an UNDEFINED tier when it is omitted from the configuration
+namespace_mapping:
+   template: "{{ project }}-{{ domain }}" # Default namespace mapping template.
```
Comment (erowan) [inline flyteadmin_config.yaml:204]: How about adding the prefix to the default template which defaults to an empty string?
"{{prefix}}{{ project }}-{{ domain }}"

& add project-prefix as an additional cmd line option to the project & file registration

Comment (wild-endeavor) [reply flyteadmin_config.yaml:204]: if the prefix isn't dynamic, why can't we statically define it in here?
```
tempate: "flyteatjpm-{{ project }}-{{ domain }}"
```

Comment (katrogan) [reply flyteadmin_config.yaml:204]: resolved offline - we'll use static prefixes
